feel-engine/src/main/scala/org/camunda/feel/impl/parser/ExpressionValidator.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.feel.impl.parser
+
+import org.camunda.feel.FeelEngine.Failure
+import org.camunda.feel.syntaxtree.{Comparison, _}
+
+class ExpressionValidator(externalFunctionsEnabled: Boolean) {
+
+  def validateExpression(expression: Exp): Option[Failure] =
+    validate(expression).headOption
+
+  private def validate(exp: Exp): List[Failure] = exp match {
+    // validate expression
+    case JavaFunctionInvocation(_, _, _) if !externalFunctionsEnabled =>
```

Comment:
Why just JavaFunctionInvocation? Or is this the only case for external function invocations in this code base?

Reply:
Yes. External functions are parsed as `JavaFunctionInvocation`. (https://github.com/camunda/feel-scala/blob/master/feel-engine/src/main/scala/org/camunda/feel/impl/parser/FeelParser.scala#L430-L434)